components/IscnUploadForm.vue

```diff
@@ -451,6 +456,106 @@ export default class UploadForm extends Vue {
     }
   }
 
+  async processEPub({ buffer, file }: { buffer: ArrayBuffer; file: File }) {
+    try {
+      const Book = ePub(buffer)
+      await Book.ready
+      const epubMetadata: any = {}
+
+      // Get metadata
+      const { metadata } = Book.packaging
+      if (metadata) {
+        epubMetadata.epubFileName = file.name
+        epubMetadata.title = metadata.title
+        epubMetadata.author = metadata.creator
+        epubMetadata.language = this.formatLanguage(metadata.language)
+        epubMetadata.description = metadata.description
+      }
+
+      // Get tags
+      const opfFilePath = await (Book.path as any).path
+      const opfContent = await Book.archive.getText(opfFilePath)
+      const parser = new DOMParser()
+      const opfDocument = parser.parseFromString(opfContent, 'application/xml')
+      const dcSubjectElements = opfDocument.querySelectorAll(
+        'dc\\:subject, subject',
+      )
+      const subjects: string[] = []
+      dcSubjectElements.forEach((element) => {
+        const subject = element.textContent
+        subject && subjects.push(subject)
+      })
+      epubMetadata.tags = subjects
+
+      // Get cover file
+      const coverUrl = (Book as any).cover
```

Comment:
Can we use documented method `[coverUrl](https://github.com/futurepress/epub.js/blob/master/documentation/md/API.md#coverurl)` instead?